sqlglot/optimizer/qualify_columns.py

```diff
@@ -776,6 +785,10 @@ def all_columns(self) -> t.Set[str]:
 
     def get_source_columns(self, name: str, only_visible: bool = False) -> t.Sequence[str]:
         """Resolve the source columns for a given source `name`."""
+        source_columns = self._get_source_columns_cache.get((name, only_visible))
```

Comment:
i like the pattern of

if ... not in ...:
    self._source_columns = ...
return self._source_column[(name, only_visible)]

Reply:
yea

Reply:
[Done](https://github.com/tobymao/sqlglot/pull/3740/commits/edcca349198e68b8c39980d09d30a3e60cd40c73)